### PR TITLE
fix(compiler): Add enum for group types. 

### PR DESCRIPTION
**Pull request by NikeNano**
**Description of your changes:**
This PR adds an ENUM type to represen group types in the DSL, see comment https://github.com/kubeflow/pipelines/pull/4199#discussion_r546868327. 

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by Ark-kun** [comment]
I wonder whether you might have missed some places.

File: sdk/python/kfp/compiler/compiler.py [hunk 1]
```diff
@@ -763,7 +770,7 @@ def _validate_exit_handler(self, pipeline):
     """
 
     def _validate_exit_handler_helper(group, exiting_op_names, handler_exists):
-      if group.type == 'exit_handler':
+      if group.type == Group.EXIT_HANDLER:
```
Comment (Ark-kun) [inline sdk/python/kfp/compiler/compiler.py:773]: Should it be `Group.EXIT_HANDLER.value`?

File: sdk/python/kfp/compiler/compiler.py [hunk 2]
```diff
@@ -38,6 +39,12 @@
 from ..dsl._pipeline_param import extract_pipelineparams_from_any, PipelineParam
 
 
+class Group(Enum):
```
Comment (Ark-kun) [inline sdk/python/kfp/compiler/compiler.py:42]: Perhaps `GroupType`?

**Comment by NikeNano** [comment]
> I wonder whether you might have missed some places.

Is this resolved through the things you mentioned @Ark-kun or is it something more that I am missing? Thanks for the feedback.

**Pull request closed without merging**